.github/workflows/pull-request.yaml

```diff
@@ -1,36 +1,43 @@
-name: Pull Request
+name: Lint, Test, and Danger
 
 on: pull_request
 
 jobs:
-  build:
+  lint-test-danger:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-      with:
-        token: '${{ secrets.PAT_DECRAPIFIER }}'
-        fetch-depth: 0
-    - name: Use Node.js 16.x
-      uses: actions/setup-node@v1
-      with:
-        node-version: 16.x
-    - name: Cache Node.js modules
-      uses: actions/cache@v2
-      with:
-        # npm cache files are stored in `~/.npm` on Linux/macOS
-        path: ~/.npm
-        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.OS }}-node-
-          ${{ runner.OS }}-
-    - run: npm ci --legacy-peer-deps
-    - run: npm run build --if-present
-    - run: npm run test:ci || touch FAIL_JOB
-    - run: npm run lint:ci || touch FAIL_JOB
-    - name: pre-danger
-      run: git config user.email "decrapifier@govspend.com" && git config user.name "Decrapifier" && git checkout ${{github.head_ref}} && git config push.default upstream && git branch -u origin/${{github.head_ref}}
-    - run: npm run danger
-      env:
-        DANGER_GITHUB_API_TOKEN: '${{ secrets.PAT_DECRAPIFIER }}'
-    - name: check-if-lint-or-test-runs-failed
-      run: 'if test -f "FAIL_JOB"; then exit 1; fi'
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PAT_DECRAPIFIER }}
+          fetch-depth: 0
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Lint
+        run: yarn run lint:ci || touch FAIL_JOB
+
+      - name: Test
+        run: yarn run test:ci || touch FAIL_JOB
+
+      - name: Setup Decrapifier user
+        run: |
+          git config user.email "decrapifier@govspend.com"
+          git config user.name "Decrapifier"
+          git config push.default upstream
+          git checkout --track origin/${{github.head_ref}}
+
+      - name: Run danger
+        run: yarn run danger ci || touch FAIL_JOB
```

Comment:
I would omit the the `|| touch FAIL_JOB` it honestly just makes it confusing when the job fails and this is last step that can fail any way so no sense in adding more complexity than there was before. 

But if you we were to keep it here too, let's update `name: check-if-lint-or-test-runs-failed` to reflect this is now included in that check too.

Reply:
Good suggestion. I removed it